File: spring-reactive-02/learn-java/src/main/java/boki/learnjava/v6_error/a_error/CharacterValidator.java
package boki.learnjava.v6_error.a_error;

import java.util.zip.DataFormatException;

public class CharacterValidator {
    public static void isAlphabeticCharacter(char ch) throws DataFormatException {
        if (!Character.isAlphabetic(ch)) {
            throw new DataFormatException("Not Alphabetic");
        }
    }
}

File: spring-reactive-02/learn-kt/src/main/kotlin/boki/learnkt/v6_error/a_error/CharacterValidator.kt
package boki.learnkt.v6_error.a_error

import java.util.zip.DataFormatException

object CharacterValidator {

    @Throws(DataFormatException::class)
    fun isAlphabeticCharacter(ch: Char) {
        if (!Character.isAlphabetic(ch.code)) {
            throw DataFormatException("Not Alphabetic")
        }
    }
}